### PR TITLE
Use pylint to improve code quality of configure.py

### PR DESCRIPTION
**Pull request by webmaster128**
As mentioned in #821, pylint is a tool that can help us identify bugs or potential problems in the configure script before they happen on the user's machine. This is especially valuable since in Python a lot of errors are not detected until the code branch is executed.

The default pylint configuration applies PEP8, so we have snake_case variable names and a bunch of other conventions.

While pylint can be annoying in some cases, in my experience it helps a lot in the longer run. And in those cases, we can silent pylint rules project global, per module or even per line.

The long .pylintrc file is more or less the default. I only applied some things that are useful for us:

- turn off full reports (reports=no)
- disable rule "missing-docstring"
- disable rule "logging-not-lazy" (because we don't care for speed)
- set line length limit to 120
- set line limit per module to 3000
- allow short variable names

I think it is another one or two hours to resolve the remaining issues, but I think for a basic setup to get the idea, this is a good first step.

### CODE REVIEW

File: configure.py [hunk 1]
```diff
@@ -2393,8 +2410,7 @@ def release_date(datestamp):
 if __name__ == '__main__':
     try:
         main()
-    except Exception as e:
-        import traceback
+    except ConfigureError as e:
```
Comment (randombit) [inline configure.py]: This last outmost `except`  should stay `Exception` - we want to capture all runtime errors here, not just ones we generated.

File: src/scripts/ci/travis/install.sh [hunk 2]
```diff
@@ -20,6 +20,9 @@ fi
 if [ "$TRAVIS_OS_NAME" = "linux" ]; then
     sudo apt-get -qq update
 
+    pip install --user pylint
+    pip3 install --user pylint
+
```
Comment (randombit) [inline src/scripts/ci/travis/install.sh]: The pylint run should go into it's own Travis build without a binary, like docs target does.

That way when `pip pylint` install breaks some day, it just fails the pylint sub build, not all Travis builds. And when it fails it is obvious what is breaking.

File: configure.py (hunk 1 above)
Comment (webmaster128) [inline configure.py]: Do we really want to catch stuff we are not prepared for? I think everything that is thrown without someone thinking about the consequences should crash the program with stack trace.

File: src/scripts/ci/travis/install.sh (hunk 2 above)
Comment (webmaster128) [inline src/scripts/ci/travis/install.sh]: Good idea!

File: configure.py (hunk 1 above)
Comment (randombit) [inline configure.py]: Capturing and logging the stack trace regardless of what happened during `main` is exactly what this top level block is for.

**Comment by webmaster128** [comment]
Good, htanks for feedback. I'll implement it

**Comment by webmaster128** [comment]
Okay, I think I have everything in there now

**Pull request merged**